src/main/kotlin/blackjack/domain/result/Rank.kt

```diff
@@ -0,0 +1,21 @@
+package blackjack.domain.result
+
+import blackjack.domain.participantion.Player
+import blackjack.domain.participantion.Price
+
+sealed class Rank(
+    private val players: List<Player>
+) {
+
+    fun exist(name: String): Boolean =
+        players.any { player -> player.name == name }
+
+    fun getTotalPrice(): Price =
+        players.fold(Price(0)) { acc, player ->
+            val price = player.price
+
+            acc + price
+        }
```

Comment:
```suggestion
    fun getTotalPrice(): Price =  Price(players.sumOf { it.priceAmount })
```
이렇게해도 문제 없을 것 같아보여요 :)